### ISSUE TITLE
[Order page] The stock location column shouldn't be displayed if no location has been indicated

### ISSUE THREAD

**Issue by colinegin** [PrestaShop/PrestaShop#18104]
Currently on the new order page for 1.7.7, the stock location column is displayed even though no stock location has been indicated for the product. This occurs in :
- the order preview from the order listing
- the cancel product block
- the standard refund block
- the partial refund block
- the return product block

It should not be displayed when the field is empty. The column should be displayed only if one product of the order has a stock location value.

**Comment by matks** [comment]
Hi @colinegin are you sure we want this feature ?

There is no complex logic in it, but I can guess one technical danger: if we hide/show stock location column whether it's needed or not, it means the blocks can have either X or X+1 columns.

So any UX improvement, design improvement, responsive improvement we bring to this page will require to think "what happens when column stock location is hidden ? what happens when column is not hidden ?" (and then it must work for all sizes of screens, all supported browsers ...)

Basically it will x2 the number of usecases to consider. We can do it but I wanted to confirm you really want this increase of complexity in the design of this already heavy page.

**Comment by colinegin** [comment]
ping @TristanLDD @MatShir your call ;)

**Comment by Progi1984** [comment]
I have added this in the new order page. May be that have been removed with merge (Poke @jolelievre or @matthieu-rolland )

**Comment by TristanLDD** [comment]
@matks I get your point, but for the merchants who doesn't use this feature, having the column hidden would result in a more confortable reading of the table on a daily basis, and contribute to make the design less heavy. These tables are potentially the ones which will be used the most often in the BO. To me, improving the user experience with details like this is enough to justify handling that additional layer of complexity.

**Comment by Progi1984** [comment]
Fixed by #18347
